src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -674,7 +674,28 @@ void InsertLocalDeclarationStatement(TLocalDeclarationStatementSyntax declaratio
                     // If the insertion node is being removed, keep the leading trivia with the new declaration.
                     if (nodesToRemove.Contains(insertionNode) && !processedNodes.Contains(insertionNode))
                     {
-                        declarationStatement = declarationStatement.WithLeadingTrivia(insertionNode.GetLeadingTrivia());
+                        // Fix 48070 - The Leading Trivia of the insertion node needs to be filtered
+                        // to only include trivia after Directives (if there are any)
+                        var leadingTrivia = insertionNode.GetLeadingTrivia();
+                        var leadingTriviaContainsDirective = leadingTrivia.Any(t => t.IsDirective);
+                        if (leadingTriviaContainsDirective)
+                        {
+                            var lastDirectiveTriviaIndex = 0;
+                            for (var i = 0; i < leadingTrivia.Count; i++)
+                            {
+                                if (leadingTrivia[i].IsDirective)
+                                {
+                                    lastDirectiveTriviaIndex = i;
+                                }
+                            }
+
+                            declarationStatement = declarationStatement.WithLeadingTrivia(insertionNode.GetLeadingTrivia().Skip(lastDirectiveTriviaIndex + 1));
```

Comment:
this feels like this could be simplified to just initially do:

```
var lastDirective = leadingTrivia.LastOrDefault(t => t.IsDirective);
if (lastDirective == default)
   // normal case
else
   // consume trivia after last directive
```